### DISCUSSION TITLE
Static file name for current wallpaper

### DISCUSSION THREAD

**Discussion by polygonnedpotato** [linuxmint/discussions#348] [Ideas]
Currently, Cinnamon stores the current wallpaper in a cache at `~/.cache/wallpaper/`, but the file name changes every time. In Windows, an easy way to acquire the current wallpaper is to used the path `%AppData%\Microsoft\Windows\Themes\TranscodedWallpaper`.

I propose doing a similar feature, having a static path to a cached copy of the current wallpaper, so that programs could easily inherit the wallpaper, such as terminals or chat clients.

**Comment by mtwebster** [comment]
Rather than a static path, you can find the current background file in Cinnamon:
```
gsettings get org.cinnamon.desktop.background picture-uri
```
Other desktops may use other keys - MATE is `org.mate.background picture-filename` I think, Gnome some others use `org.gnome.desktop.background picture-uri`.  XFCE has their own settings format, so I'm not sure for them.

You can browse this stuff using the dconf-editor utility.

This is also available programmatically (such as from a python script), using Gio.Settings.
